### PR TITLE
Migrate to Github Actions

### LINKED ISSUE #74: Migrate to Github Actions
Since Travis do not support Open Source anymore we need to move the C# SDK on Github Actions

Move CI to cross-repo doc build & deploy, like it has been done in the [Dart SDK](https://github.com/kuzzleio/sdk-dart/blob/3-dev/.github/workflows/push_master.workflow.yml)

### Branches
- [ ] `master`
- [ ] `2-dev`

### PR DESCRIPTION
**Pull request by rolljee**
Update GH actions to make it more standard accros our repo

### CODE REVIEW

**Review by Aschen — APPROVED** [review]

File: .github/actions/deploy/action.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+description: Run Deploy
+runs:
+  using: "composite"
+  steps:
+    - name: Run restore
+      run: dotnet restore
+      shell: bash
+    - name: Run Deploy
+      run: |
+        dotnet build $TRAVIS_BUILD_DIR/Kuzzle/Kuzzle.csproj -c Release
```
Comment (rolljee) [inline .github/actions/deploy/action.yml]: This can't works in Github action need changes

File: .github/actions/deploy/action.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+        dotnet build $TRAVIS_BUILD_DIR/Kuzzle.Tests/Kuzzle.Tests.csproj
```
Comment (rolljee) [inline .github/actions/deploy/action.yml]: This can't works in Github action need changes

File: .github/actions/unit-tests/action.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+description: Run Unit Tests
+runs:
+  using: "composite"
+  steps:
+    - name: Run restore
+      run: dotnet restore
+      shell: bash
+    - name: Run tests
+      run: |
+        dotnet test /p:CollectCoverage=true /p:CoverletOutputFormat=lcov /p:CoverletOutput=$TRAVIS_BUILD_DIR/lcov.info
```
Comment (rolljee) [inline .github/actions/unit-tests/action.yml]: This can't work in github action

**Comment by Aschen** [comment]
@rolljee I think you need to remove the `.travis-ci.yml` file

**Comment by rolljee** [comment]
> @rolljee I think you need to remove the `.travis-ci.yml` file

File was already remove, it was under settings > branches > protection for each branches, where you change required jobs

**Review by alexandrebouthinon — APPROVED** [review]
You should merge `doc-prod.yml` into `push_master.yml` to avoid workflow condition duplication but it's ok

File: .github/workflows/pull_request.workflow.yml [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+            ${{ runner.os }}-build-${{ env.cache-name }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
+      - uses: actions/setup-node@v1.4.4
+        with:
+          node-version: "12"
+      - uses: ./.github/actions/build-docs
+
+  unit-tests:
+    name: Unit Tests csharp ${{ matrix.dotnet }}
```
Comment (alexandrebouthinon) [inline .github/workflows/pull_request.workflow.yml]:
```suggestion
    name: Unit Tests - .NET v${{ matrix.dotnet }}
```

File: .github/workflows/pull_request.workflow.yml [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup dotnet
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: ${{ matrix.dotnet }}
+      - uses: ./.github/actions/unit-tests
+
+  snippets-tests:
+    name: Snippet Tests csharp ${{ matrix.dotnet }}
```
Comment (alexandrebouthinon) [inline .github/workflows/pull_request.workflow.yml]:
```suggestion
    name: Snippet Tests - .NET v${{ matrix.dotnet }}
```

File: .github/workflows/push_dev.workflow.yml [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup dotnet
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: ${{ matrix.dotnet }}
+      - uses: ./.github/actions/unit-tests
+
+  snippets-tests:
+    name: Snippet Tests csharp ${{ matrix.dotnet }}
```
Comment (alexandrebouthinon) [inline .github/workflows/push_dev.workflow.yml]:
```suggestion
    name: Snippet Tests - .NET v${{ matrix.dotnet }}
```

File: .github/workflows/push_dev.workflow.yml [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+
+on:
+   push:
+     branches:
+       - 2-dev
+  
+
+jobs:
+  unit-tests:
+    name: Unit Tests csharp ${{ matrix.dotnet }}
```
Comment (alexandrebouthinon) [inline .github/workflows/push_dev.workflow.yml]:
```suggestion
    name: Unit Tests - .NET v${{ matrix.dotnet }}
```

File: .github/workflows/push_master.workflow.yml [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+
+on:
+   push:
+     branches:
+       - master
+
+
+jobs:
+  unit-tests:
+    name: Unit Tests csharp ${{ matrix.dotnet }}
```
Comment (alexandrebouthinon) [inline .github/workflows/push_master.workflow.yml]:
```suggestion
    name: Unit Tests - .NET v${{ matrix.dotnet }}
```

File: .github/workflows/push_master.workflow.yml [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup dotnet
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: ${{ matrix.dotnet }}
+      - uses: ./.github/actions/unit-tests
+
+  snippets-tests:
+    name: Snippet Tests csharp ${{ matrix.dotnet }}
```
Comment (alexandrebouthinon) [inline .github/workflows/push_master.workflow.yml]:
```suggestion
    name: Snippet Tests - .NET v${{ matrix.dotnet }}
```

**Review by alexandrebouthinon — APPROVED** [review]
nit

**Comment by rolljee** [comment]
> You should merge `doc-prod.yml` into `push_master.yml` to avoid workflow condition duplication but it's ok

As you mentioned earlier I merged those 2 files, and found some mistakes along the way 👍

File: .github/workflows/push_dev.workflow.yml [hunk 10]
```diff
@@ -0,0 +1,77 @@
+name: Deploy Docs Dev
```
Comment (alexandrebouthinon) [inline .github/workflows/push_dev.workflow.yml]:
```suggestion
name: Dev branch
```

File: .github/workflows/pull_request.workflow.yml [hunk 11]
```diff
@@ -0,0 +1,88 @@
+name: Pull Request workflow
```
Comment (alexandrebouthinon) [inline .github/workflows/pull_request.workflow.yml]:
```suggestion
name: Pull Request

File: .github/workflows/push_master.workflow.yml [hunk 12]
```diff
@@ -0,0 +1,96 @@
+name: Deploy Prod to NuGet
```
Comment (alexandrebouthinon) [inline .github/workflows/push_master.workflow.yml]:
```suggestion
name: Main branch
```

**Review by alexandrebouthinon — APPROVED** [review]
Nit, merge when applied

**Pull request merged**